merlin/models/tf/transforms/sequence.py

```diff
@@ -219,7 +219,7 @@ def call(
         self._check_seq_inputs_targets(inputs)
 
         # Shifts the target column to be the next item of corresponding input column
-        new_target = inputs[self.target_name][:, 1:]
+        new_target = inputs[self.target_name][:, -1:]
```

Comment:
The tranform `SequencePredictNext` aims to perform a sliding window prediction of size 1 (we referred to it as Causal Language modeling in T4Rec to align with the NLP domain). More specifically, the session-based model will use the hidden representation at position N to predict the target at position N+1. So each input row in the batch will be linked to multiple targets (from position 2 up to last non-padded position)

If you want to predict only the last item of the sequence you should use `SequencePredictLast` indead.

Reply:
Thanks for the explanation. I think I got confused here between predict next and predict last.